pkg/kafka/encoding.go

```diff
@@ -188,3 +194,55 @@ func (d *Decoder) DecodeWithoutLabels(data []byte) (logproto.Stream, error) {
 func sovPush(x uint64) (n int) {
 	return (math_bits.Len64(x|1) + 6) / 7
 }
+
+// EncodeStreamMetadata encodes the stream metadata into a Kafka record
+// using the tenantID as the key and partition as the target partition
+func EncodeStreamMetadata(partition int32, topic string, tenantID string, streamHash uint64) *kgo.Record {
+	// Validate stream hash
+	if streamHash == 0 {
+		return nil
+	}
+
+	metadata := logproto.StreamMetadata{
+		StreamHash: streamHash,
+	}
+
+	// Encode the metadata into a byte slice
+	value, err := metadata.Marshal()
+	if err != nil {
+		// Since we're in a function that returns a *kgo.Record, we can't return an error.
+		// The best we can do is return nil which will fail later.
+		return nil
+	}
+
+	return &kgo.Record{
+		Key:       []byte(tenantID),
+		Value:     value,
+		Partition: partition,
+		Topic:     MetadataTopicFor(topic),
+	}
+}
+
+// DecodeStreamMetadata decodes a Kafka record into a StreamMetadata.
+// It returns the decoded metadata and any error encountered.
+func DecodeStreamMetadata(record *kgo.Record) (*logproto.StreamMetadata, error) {
+	if record == nil {
```

Comment:
@periklis Do we need these two checks? It should be expected that the caller passes a non-nil `record` that also has a non-nil `record.Value`?

Reply:
Agree with that we can drop them now.